### DISCUSSION TITLE
Multiple openApi documents

### DISCUSSION THREAD

**Discussion by Wulfheart** [dedoc/scramble#174] [Ideas]
Hey,

it would be really great if it was possible to generate multiple OpenAPI documents from one codebase.

**Comment by romalytvynenko** [comment]
@Wulfheart would you mind sharing why it would be great?

**Reply by romalytvynenko** [reply]
I mean, what is your use case 🤔

**Reply by Wulfheart** [reply]
I want to provide multiple APIs in one application and they have different use cases.

**Reply by romalytvynenko** [reply]
I see, thanks.

Please, tell more about how you set it up and how it looks like in your app.

**Comment by tontonsb** [comment]
Hey, we have a similar need.

We have multiple APIs for different audiences, so it would ideally be like

- `/api/v2/*` goes to `/docs/api/v2` docs and has one set of permissions and other config
- `/public-api/*` goes to `/docs/api/public` docs and has public access and different other config

Currently we can only create multiple setups by sequentially changing the config file, generating the openAPI doc and saving it.

As it is a bit of a custom feature, it might make sense to let us use the generator and pass custom config. The default route (embedded in package) could be something like this:

```php
    Route::get('docs/api.json', function (Dedoc\Scramble\Generator $generator) {
        return $generator(config('scramble'));
    })->name('scramble.docs.index');

    Route::view('docs/api', 'scramble::docs')->name('scramble.docs.api');
```

And we could then define our own routes with custom config:

```php
    Route::get('docs/api.json', function (Dedoc\Scramble\Generator $generator) {
        return $generator([
            'api_path' => 'api/v77',
            'api_domain' => null,
        
            'info' => [
                'version' => env('API_VERSION', '77.1.500'),
                'description' => '',
            ],
            'servers' => null,
            'middleware' => [
                'web',
                'boostMemLimit:1024M',
            ],
        ]);
    })->name('apidocs.v7');

    Route::view('docs/api', 'scramble::docs', ['openapi_path' => route('apidocs.v7')])->name('apigui.v7');
```

**Reply by romalytvynenko** [reply]
That's what I needed. Thank you. Will plan the feature

**Reply by tontonsb** [reply]
I just noticed this probably overlaps with #162

**Comment by romalytvynenko** [comment]
@tontonsb @Wulfheart 

Released in `v0.10.*`! 

https://scramble.dedoc.co/usage/multiple-docs

**Reply by tontonsb** [reply]
Thanks!
